config.toml

```diff
@@ -33,6 +33,8 @@ enableGitInfo = true
 # Hindi is disabled because it's currently in development.
 disableLanguages = ["hi", "no"]
 
+rssLimit = 10
```

Comment:
I don't think it would be that puzzling, since almost all feeds are limited to some extent. But if you mean I add the feed 30 days after the release, and then I only get 10 of the 35 articles about the new release, I definitely don't think that'd be puzzling, since I probably just saw the list off all 30 articles in the blog while I was clicking the RSS link, so I clearly see that it's the most recent 10. And since I found the RSS link by looking at the blog on the web, I necessarily know how to see earlier articles if I want them.

This may be a matter of perspective, but I think that the main use case of RSS is almost always to subscribe. That is, I'm using it to pick up the last article or two every day, or every hour, or every 10 min. I'm not using it to populate a local copy of the last year's worth of posts. I don't really need that. Or at least, I don't need it nearly as much as I need to subscribe.

Making the images smaller might be a good idea for other reasons, but in this case I'm only thinking about the size of the feed as plain text. 800kb isn't very big for one feed, but I might have 200 feeds. I don't think it's a good idea to worsen the experience for subscribers and heavy users for the sake of people who want to get a local copy of the blog in their reader. However, if you think the latter group are important to cater to, I would think having some extra feed urls would be the way to do it. You have the main feed for the primary use case (subscribers keeping up to date on the latest 1-2 articles) and then a link to a page of alternate feeds, such as a feed of the entire blog since the beginning of time, or other formats (there's this jsonfeed thing too now apparently), or other things I can't think of.

But I would just do the 1 feed for the 1 case that IMO actually matters if it was me. Or at least, that's what I'd do _first_.

Reply:
For what it's worth, the feed is 200KiB gzipped. https://k8s.io/blog/ is 800KiB, too.

OK, I'm convinced.

```suggestion
# Limit number of blog articles included in the syndication feed
[services.rss]
limit = 10
```

If folks complain about the limited number of articles in the feed, we can provide a separate “last 2 years” feed or similar.

Reply:
I still think that smaller images would shrink the feed more, mind. The “plain text” of the feed is actually XML with embedded images.

Reply:
By "embedded images" do you mean html links to images, or something which directly encodes images in text, like svg or whatever? If the former, I assume we just have to live with that. If the latter, that could be a concern, but while I can't say I looked through the feed text super-carefully, I don't think I noticed any substantial amount of that. In any case, even if it was there, it's not obvious what we could do about it, or what the consequences of these options might be.

In any case, as I say, I didn't notice a whole lot of either kind of image. There were some big blocks of what I think where html code blocks which had been escaped, which do contribute nontrivial bloat. But once again I'm not totally sure we have easy options for getting rid of that bloat with a 100% guarantee of no unintended consequences. And there are plenty of articles which are bigger than the articles "bloated" by code blocks which are just big with content, like the Box/Laverack interview. So if you had a bunch of articles like that you'd be paying for the complexity of whatever you did to reduce the bloat from code blocks, but having just as bloated a feed.

This is why I wanted to solve the issue by making the feed smaller.